### ISSUE TITLE
Adding a NULL time fails even with default

### ISSUE THREAD

**Issue by jyutzler** [ngageoint/geopackage-java#31]
Please fill out as much known and relevant information as possible.

#### Version Information:

  * GeoPackage Java Version: 1.2.2
  * GeoPackage Java Source: (e.g. Central Repository, Release, Source Code Build) Local source code build after cloning trunk from GitHub
  * IDE Name & Version: Eclipse 4.6.3
  * Maven Version: (mvn -version) 3.3.9
  * Java Version: (java -version) 1.8.0_72
  * Platform & OS: MAC OSX 10.11.6
  * Other Relevant Libraries:

#### Steps to reproduce
In file /geopackage/src/test/java/mil/nga/geopackage/test/features/columns/GeometryColumnsUtils.java comment out line 530 in order to test the addition of a row to gpkg_contents with the default date.
> //			contents.setLastChange(new Date());

#### Expected Results:

  * Row inserted properly, no SQL errors

#### Observed Results:

SQL error on insert (NULL constraint violated)
(Always)

#### Output:

```
Running mil.nga.geopackage.test.features.columns.GeometryColumnsCreateTest
Tests run: 4, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.465 sec <<< FAILURE!
testCreate(mil.nga.geopackage.test.features.columns.GeometryColumnsCreateTest)  Time elapsed: 0.104 sec  <<< ERROR!
java.sql.SQLException: Unable to run insert stmt on object mil.nga.geopackage.core.contents.Contents@1dc76ef1: INSERT INTO `gpkg_contents` (`table_name` ,`data_type` ,`identifier` ,`description` ,`last_change` ,`min_x` ,`min_y` ,`max_x` ,`max_y` ,`srs_id` ,`srs_id` ) VALUES (?,?,?,?,?,?,?,?,?,?,?)
	at com.j256.ormlite.misc.SqlExceptionUtil.create(SqlExceptionUtil.java:25)
	at com.j256.ormlite.stmt.mapped.MappedCreate.insert(MappedCreate.java:137)
	at com.j256.ormlite.stmt.StatementExecutor.create(StatementExecutor.java:458)
	at com.j256.ormlite.dao.BaseDaoImpl.create(BaseDaoImpl.java:328)
	at mil.nga.geopackage.core.contents.ContentsDao.create(ContentsDao.java:79)
	at mil.nga.geopackage.test.features.columns.GeometryColumnsUtils.testCreate(GeometryColumnsUtils.java:541)
	at mil.nga.geopackage.test.features.columns.GeometryColumnsCreateTest.testCreate(GeometryColumnsCreateTest.java:83)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
Caused by: org.sqlite.SQLiteException: [SQLITE_CONSTRAINT_NOTNULL]  A NOT NULL constraint failed (NOT NULL constraint failed: gpkg_contents.last_change)
	at org.sqlite.core.DB.newSQLException(DB.java:909)
	at org.sqlite.core.DB.newSQLException(DB.java:921)
	at org.sqlite.core.DB.execute(DB.java:825)
	at org.sqlite.core.DB.executeUpdate(DB.java:863)
	at org.sqlite.jdbc3.JDBC3PreparedStatement.executeUpdate(JDBC3PreparedStatement.java:99)
	at com.j256.ormlite.jdbc.JdbcDatabaseConnection.insert(JdbcDatabaseConnection.java:183)
	at com.j256.ormlite.stmt.mapped.MappedCreate.insert(MappedCreate.java:91)
	... 38 more

Running mil.nga.geopackage.test.features.columns.GeometryColumnsImportTest
Tests run: 6, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.053 sec <<< FAILURE!
testCreate(mil.nga.geopackage.test.features.columns.GeometryColumnsImportTest)  Time elapsed: 0.017 sec  <<< ERROR!
java.sql.SQLException: Unable to run insert stmt on object mil.nga.geopackage.core.contents.Contents@6cb62e90: INSERT INTO `gpkg_contents` (`table_name` ,`data_type` ,`identifier` ,`description` ,`last_change` ,`min_x` ,`min_y` ,`max_x` ,`max_y` ,`srs_id` ,`srs_id` ) VALUES (?,?,?,?,?,?,?,?,?,?,?)
	at com.j256.ormlite.misc.SqlExceptionUtil.create(SqlExceptionUtil.java:25)
	at com.j256.ormlite.stmt.mapped.MappedCreate.insert(MappedCreate.java:137)
	at com.j256.ormlite.stmt.StatementExecutor.create(StatementExecutor.java:458)
	at com.j256.ormlite.dao.BaseDaoImpl.create(BaseDaoImpl.java:328)
	at mil.nga.geopackage.core.contents.ContentsDao.create(ContentsDao.java:79)
	at mil.nga.geopackage.test.features.columns.GeometryColumnsUtils.testCreate(GeometryColumnsUtils.java:541)
	at mil.nga.geopackage.test.features.columns.GeometryColumnsImportTest.testCreate(GeometryColumnsImportTest.java:87)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
Caused by: org.sqlite.SQLiteException: [SQLITE_CONSTRAINT_NOTNULL]  A NOT NULL constraint failed (NOT NULL constraint failed: gpkg_contents.last_change)
	at org.sqlite.core.DB.newSQLException(DB.java:909)
	at org.sqlite.core.DB.newSQLException(DB.java:921)
	at org.sqlite.core.DB.execute(DB.java:825)
	at org.sqlite.core.DB.executeUpdate(DB.java:863)
	at org.sqlite.jdbc3.JDBC3PreparedStatement.executeUpdate(JDBC3PreparedStatement.java:99)
	at com.j256.ormlite.jdbc.JdbcDatabaseConnection.insert(JdbcDatabaseConnection.java:183)
	at com.j256.ormlite.stmt.mapped.MappedCreate.insert(MappedCreate.java:91)
	... 38 more
```

#### Relevant Code:

```java
// Code to reproduce the problem?
```

#### Test Files:

  * Links to any files needed for testing?

#### Additional Information:

  * Any additional configuration, data, or information that might help with the issue?

# I need to know whether the problem is with the default mechanism or the test or the library

**Comment by bosborn** [comment]
Good find.  Problem is in the default ORM configurations within the library.

Fixed on develop branch and will release in the next versions:
Core: https://github.com/ngageoint/geopackage-core-java/commit/c85de51a2a7383e230addac0c2d75672e1983156
Java: https://github.com/ngageoint/geopackage-java/commit/64f4fd1927d6df9a80205caf85f62d0e9801a352
Android: https://github.com/ngageoint/geopackage-android/commit/2bdb27f7e445eb16a3c77a2007d62daac22c7ec9
Android Map: https://github.com/ngageoint/geopackage-android-map/commit/ba67daf20150ff9eb5d39a2691c47291ad467c30

Will leave ticket open until merged and released.

**Comment by bosborn** [comment]
fixed as part of release [1.2.2](https://github.com/ngageoint/geopackage-java/releases/tag/1.2.2)
